src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -225,52 +238,28 @@ const IndexPatternEditorFlyoutContentComponent = ({
       let newRollupIndexName: string | undefined;
 
       const fetchIndices = async (query: string = '') => {
+        const currentRequestRef = {};
```

Comment:
I usually use a counter for this purpose and simply increment it.

```
const currentLoadingMatchedIndicesRef = useRef(0);

...

const fetchIndices = async (query: string = '') => {
  const idx = ++currentLoadingMatchedIndicesRef.current;

  ...

  if (idx === currentLoadingMatchedIndicesRef.current) {
    // update states
  }
}
```

Reply:
looks good, easier to understand then the trick with ref to an object